oe_theme.theme

```diff
@@ -521,120 +521,163 @@ function oe_theme_preprocess_field(&$variables, $hook): void {
+  for (; $i <= $pager_current + $pager_offset && $i <= $pager_last; $i++) {
+    $label = (string) $i;
+    if ($i == $pager_current) {
+      $ecl_items[] = [
+        'type' => 'current',
+        'label' => $label,
+        'aria_label' => t('Page @number', ['@number' => $label]),
+      ];
+    }
+    else {
+      // Skip first or last page.
+      if ($i == 1 || $i == $pager_max) {
         continue;
       }
-      $link = [];
-      $link['aria_label'] = t('Go to page @number', ['@number' => $page_key]);
-      $link['link']['link'] = [
-        'path' => $page_item['href'],
-        'label' => (string) $page_key,
+      $options = [
+        'query' => pager_query_add_page($parameters, $element, $i - 1),
+      ];
+      $ecl_items[] = [
+        'link' => [
+          'link' => [
+            'path' => Url::fromRoute($route_name, $route_parameters, $options),
+            'label' => $label,
+          ],
+        ],
+        'aria_label' => t('Go to page @number', ['@number' => $i]),
       ];
-      $variables['ecl_items'][] = $link;
     }
   }
 
-  // Generate next ellipses if needed.
-  if (isset($variables['ellipses']['next']) && $variables['ellipses']['next']) {
-    $ellipsis = [
-      'type' => 'ellipsis',
-      'label' => '...',
-    ];
-    $variables['ecl_items'][] = $ellipsis;
+  // Add second ellipsis.
+  if ($pager_last < $pager_max - 1) {
+    $ecl_items[] = $ellipsis;
   }
 
-  // Generate next link if it is available.
-  if (!empty($variables['items']['next']['href'])) {
-    $first_link = [];
-    $first_link['type'] = 'next';
-    $first_link['aria_label'] = t('Go to next page');
-    $label = t('Next ›');
-    if (isset($tags[3])) {
-      $label = $tags[3];
-    }
-    $first_link['link']['link'] = [
-      'path' => $variables['items']['next']['href'],
-      'label' => $label,
+  // Add last page link.
+  if ($pager_max != $pager_current) {
+    $options = [
+      'query' => pager_query_add_page($parameters, $element, $pager_max - 1),
+    ];
+    $ecl_items[] = [
+      'type' => 'last',
+      'link' => [
+        'link' => [
+          'path' => Url::fromRoute($route_name, $route_parameters, $options),
+          'label' => (string) $pager_max,
+        ],
+      ],
+      'aria_label' => t('Go to page @number', ['@number' => $pager_max]),
     ];
-    $variables['ecl_items'][] = $first_link;
   }
 
-  // Generate previous link if it is available.
-  if (!empty($variables['items']['last']['href'])) {
-    $last_link = [];
-    $last_link['type'] = 'last';
-    $last_link['aria_label'] = t('Go to last page');
-    $label = t('Last ››');
-    if (isset($tags[4])) {
-      $label = $tags[4];
-    }
-    $last_link['link']['link'] = [
-      'path' => $variables['items']['last']['href'],
-      'label' => $label,
+  // Add 'Next' navigation link.
+  if ($pager_current < $pager_max) {
+    $options = [
+      'query' => pager_query_add_page($parameters, $element, $pager_page_array[$element] + 1),
```

Comment:
`pager_query_add_page()` is deprecated function in drupal:8.8.0. Use `\Drupal\Core\Pager\PagerManagerInterface::getUpdatedParameters()` instead.